komodo/cli.py

```diff
@@ -62,190 +103,308 @@ def _print_timings(
+        outdir=download_destination,
+        pip=pip_executable,
+    )
+
+    return git_hashes
+
+
+@profile_time("Building non-pip part of komodo in workspace")
+def _make(*args, **kwargs) -> None:
+    make(*args, **kwargs)
+
+
+def build_packages_and_move_to_release_path(
+    args: KomodoNamespace, data: Data, tmp_prefix: Path
+):
     fakeroot = Path(args.release).resolve()
-    if args.build or not args.install:
-        start_time = datetime.datetime.now()
-        make(
-            args.pkgs.content,
-            args.repo.content,
-            data,
-            prefix=str(tmp_prefix),
-            dlprefix=args.downloads,
-            builddir=args.tmp,
-            jobs=args.jobs,
-            cmk=args.cmake,
-            pip=args.pip,
-            fakeroot=str(fakeroot),
-        )
-        timings.append(
-            (
-                "Building non-pip part of komodo in workspace",
-                datetime.datetime.now() - start_time,
-            ),
-        )
-        _print_timings(timings[-1])
+    _make(
+        args.pkgs.content,
+        args.repo.content,
+        data,
+        prefix=str(tmp_prefix),
+        dlprefix=args.downloads,
+        builddir=args.tmp,
+        jobs=args.jobs,
+        cmk=args.cmake,
+        pip=args.pip,
+        fakeroot=str(fakeroot),
+    )
 
-        shell(f"mv {args.release + str(tmp_prefix)} {args.release}")
-        with contextlib.suppress(OSError):
-            os.removedirs(f"{args.release + str(tmp_prefix.parent)}")
+    shell(f"mv {args.release + str(tmp_prefix)} {args.release}")
+    with contextlib.suppress(OSError):
+        os.removedirs(f"{args.release + str(tmp_prefix.parent)}")
 
-    if args.build and not args.install:
+
+def exit_on_build_only_and_no_install_run(build_mode: bool, install_mode: bool) -> None:
+    if build_mode and not install_mode:
         sys.exit(0)
 
-    create_enable_scripts(komodo_prefix=tmp_prefix, komodo_release=args.release)
-
-    releasedoc = Path(args.release) / Path(args.release)
-    with open(releasedoc, "w", encoding="utf-8") as filehandle:
-        release = {}
-        for pkg, ver in args.pkgs.content.items():
-            entry = args.repo.content[pkg][ver]
-            maintainer = args.repo.content[pkg][ver]["maintainer"]
-            if ver == LATEST_PACKAGE_ALIAS:
-                ver = latest_pypi_version(entry.get("pypi_package_name", pkg))
-            elif args.repo.content[pkg][ver].get("fetch") == "git":
-                ver = git_hashes[pkg]
-            release[pkg] = {
-                "version": ver,
+
+def generate_release_root(release_path: Path) -> Path:
+    """Append root to the temporary build dir, as we want a named root/
+    directory as the distribution root, organised under the distribution name
+
+    Args:
+        release_path (Path):
+
+    Returns:
+        release_root
+    """
+    abs_prefix = release_path.resolve()
+    return abs_prefix / "root"
+
+
+def generate_releasedoc(
```

Comment:
I'd maybe change the name to `general_release_packages`?

Reply:
Yes, works fine for me

Reply:
Sorry, I meant `generate_release_packages` :smile: